mule-apikit-metadata/src/main/java/org/mule/module/apikit/metadata/internal/amf/AmfHandler.java

```diff
@@ -57,19 +51,13 @@ public AmfHandler(final ResourceLoader resourceLoader, final Notifier notifier)
     final ParserWrapperAmf parserWrapper;
 
     try {
+      final ApiRef apiRef;
       if (APISyncUtils.isSyncProtocol(apiDefinition)) {
-        Environment environment = DefaultEnvironment.apply();
-        environment = environment.add(new ResourceLoaderWrapper(resourceLoader, apiDefinition));
-        parserWrapper = ParserWrapperAmf.create(apiDefinition, environment, true);
+        apiRef = ApiRef.create(apiDefinition, resourceLoader);
       } else {
-
-        final URI uri = resourceLoader.getResource(apiDefinition);
-        if (uri == null) {
-          notifier.error(format("API definition '%s' not found using AMF parser.", apiDefinition));
-          return empty();
-        }
-        parserWrapper = ParserWrapperAmf.create(uri, true);
+        apiRef = ApiRef.create(apiDefinition);
```

Comment:
Metadata always has a ResourceLoader provided by DataSense even when it is not SyncProtocol